Source/Renderer/ShaderProgram.js

```diff
@@ -27,7 +29,51 @@ define([
     /**
      * @private
      */
+
+    function extractUniforms(shaderText) {
+        // TODO: regex doesn't work for inline structs
+        var uniformNames = [];
+        var uniformLines = shaderText.match(/uniform[\w\s]*(?=[=\[;])/g);
+        if (uniformLines) {
+            for (var i = 0; i < uniformLines.length; i++) {
+                var line = uniformLines[i].trim();
+                var name = line.slice(line.lastIndexOf(' ') + 1);
+                uniformNames.push(name);
+            }
+        }
+        return uniformNames;
+    }
+
     var ShaderProgram = function(options) {
+        // If a uniform exists in both the vertex and fragment shader but with different precision qualifiers,
+        // give the fragment shader uniform a different name. This fixes shader compilation errors on devices
+        // that only support mediump in the fragment shader.
+        var vertexShaderText = options.vertexShaderText;
+        var fragmentShaderText = options.fragmentShaderText;
+        var duplicateUniformNames = {};
+
+        if (!ContextLimits.highpSupported) {
+            var vertexShaderUniforms = extractUniforms(vertexShaderText);
+            var fragmentShaderUniforms = extractUniforms(fragmentShaderText);
+
+            // Find uniforms used in both
+            var i, j;
+            var uniformName;
+            var duplicateName;
+            for (i = 0; i < vertexShaderUniforms.length; i++) {
+                for (j = 0; j < fragmentShaderUniforms.length; j++) {
+                    if (vertexShaderUniforms[i] === fragmentShaderUniforms[j]) {
+                        uniformName = vertexShaderUniforms[i];
+                        duplicateName = uniformName + "_f";
+                        // Update fragmentShaderText with renamed uniforms
+                        var re = new RegExp(uniformName + "\\b", "g");
+                        fragmentShaderText = fragmentShaderText.replace(re, duplicateName);
```

Comment:
It doesn't matter here, but whenever we add/remove new lines in GLSL, we use `#line` (or add new lines) to preserve the line numbers for errors/warnings ([for example](https://github.com/lilleyse/cesium/blob/uniform-mediump/Source/Renderer/ShaderSource.js#L163)).